rocAL/rocAL_pybind/amd/rocal/pipeline.py

```diff
@@ -215,18 +215,18 @@ def copyToExternalTensorNCHW(self, array,  multiplier, offset, reverse_channels,
                                         multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0), self._output_memory_type)
             elif tensor_dtype == types.FLOAT16:
                 b.rocalCupyToTensor16(self._handle, ctypes.c_void_p(array.ctypes.data), types.NCHW,
-                                       multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0), self._output_memory_type)
+                                        multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0), self._output_memory_type)
     def GetOneHotEncodedLabels(self, array, device):
         if device=="cpu":
             if (isinstance(array,np.ndarray)):
                 b.getOneHotEncodedLabels(self._handle, array.ctypes.data_as(ctypes.c_void_p), self._numOfClasses, 0)
             else: #torch tensor
                 return b.getOneHotEncodedLabels(self._handle, ctypes.c_void_p(array.data_ptr()), self._numOfClasses, 0)
-        if device=="gpu":
+        else:
             if (isinstance(array,cp.ndarray)):
-                b.getOneHotEncodedLabels(self._handle, array.data.ptr, self._numOfClasses, 1)
+                b.getCupyOneHotEncodedLabels(self._handle, array.data.ptr, self._numOfClasses, 1)
             else: #torch tensor
-                return b.getOneHotEncodedLabels(self._handle, ctypes.c_void_p(array.data_ptr()), self._numOfClasses, 1)
+                return b.getCupyOneHotEncodedLabels(self._handle, ctypes.c_void_p(array.data_ptr()), self._numOfClasses, 1)
```

Comment:
@swetha097 Why are we changing the torch part to use cupy as well? I thought only tensorflow and bare metal GPU uses the CuPy flow and torch uses it's own tensor. Can you check on this?

Reply:
Changed it back to using getOneHotEncodedLabels API for the torch GPU tensors.